pyproject.toml

```diff
@@ -8,18 +8,19 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = "3.9.*"
-pyspark = {version = "3.3.*", extras = ["sql"]}
-dask = {version = "2022.10.*", extras = ["complete"]}
-vaex = "4.14.*"
-polars = "0.14.28"
-duckdb = "0.5.1"
+python = "^3.11"
```

Comment:
Would you be able to change all the dependencies so that they're hard pinned to the bug version rather than lower-bound constrained via the `^`? Otherwise, these benchmarks will change whenever the code is rerun and a new version is released.

Reply:
I'm a bit confused what you are asking for here. The `poetry.lock` file has pinned versions of the direct dependencies and all transitive dependencies.  It's committed to the repo, and used in the Dockerfile before creating the `requirements.txt`.   If someone does `poetry install` it will source the pinned versions in the lockfile.  If there are new releases of any of the dependencies with relaxed constraints in the `pyproject.toml` they will be ignored because of the existence of the `poetry.lock` file.  At some later date you can opt to refresh the lockfile to use the latest versions of all the dependencies that still satisfy the constraints listed in `pyproject.toml` using `poetry update`.That will not change the constraints in `pyproject.toml`, only re-generate the lockfile if there are newer versions that still satisfy them.  The lockfile also attempts to be operating system and cpu architecture independent for additional reproducibility (though it doesn't always succeed).

Reply:
You're totally right :man_facepalming: I think I've been misunderstanding the lock file this whole time... Please ignore me!